### PR TITLE
[#153727841:Feature] Implemented missing front tests

### PR DESCRIPTION
**Pull request by samachola**
## PR Status
**READY**

## Story ID
#153727841

## Added Migrations
NO

## Description
This PR is intended to implement the missing front tests. 

## New Packages Added
None

## Related PRs 
None

## Todos
- [ ] Reviewed and approved by a TTL

## Impacted Areas in the application
None

## Known Issues (optional)
None

### CODE REVIEW

File: hc/front/tests/test_add_channel.py [hunk 1]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
             self.assertContains(r, "Integration Settings", status_code=200)
 
     ### Test that the team access works
+    def test_teamaccess_works(self):
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: try using `reverse`

File: hc/front/tests/test_add_channel.py [hunk 2]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
...
+        url = "/integrations/add/"
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: Try the endpoint `hc-add-channel` with reverse

File: hc/front/tests/test_add_channel.py [hunk 3]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
...
+        form = {"kind": "email", "value": "alice@example.com"}
+
+        self.client.login(username="alice@example.org", password="password")
+        self.client.post(url, form)
+        alice_channels = Channel.objects.filter(user=self.alice)
+        self.client.logout()
+
+        self.client.login(username="bob@example.org", password="password")
+        r = self.client.get("/integrations/")
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: use `reverse`
You can read about it from the Django documentation

File: hc/front/tests/test_add_channel.py [hunk 4]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
...
+        self.assertEquals(r.status_code, 200)
+
+
+
     ### Test that bad kinds don't work
+    def test_bad_kinds_dont_work(self):
+        # Bad kinds means that a user can't add unsupported intergrations.
+        self.client.login(username="alice@example.org")
+        bad_kinds = ("whatsapp", "facebook", "linkedin")
+        for frag in bad_kinds:
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: Use a meaningful variable name

File: hc/front/tests/test_add_channel.py [hunk 5]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
...
+            url = "/integrations/add_%s/" % frag
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: use `reverse`

File: hc/front/tests/test_add_check.py [hunk 6]
```diff
@@ -12,3 +12,18 @@ def test_it_works(self):
...
     ### Test that team access works
+    def test_team_access_works(self):
+        
+        self.client.login(username="bob@example.org", password="password")
+        self.client.get("/checks/")
+        assert Check.objects.count() == 0
+        self.client.logout()
+        
+        self.client.login(username="alice@example.org", password="password")
+        url = "/checks/add/"
```
Comment (johnseremba) [inline hc/front/tests/test_add_check.py]: refer to the above comments

File: hc/front/tests/test_add_pushbover.py [hunk 7]
```diff
@@ -38,3 +38,13 @@ def test_it_validates_nonce(self):
...
     ### Test that pushover validates priority
+    def test_pushover_validates_priority(self):
+        self.client.login(username="alice@example.org", password="password")
+
+        session = self.client.session
+        session["po_nonce"] = "n"
+        session.save()
+
+        params = "pushover_user_key=a&nonce=n&prio=13"
+        r = self.client.get("/integrations/add_pushover/?%s" % params)
```
Comment (johnseremba) [inline hc/front/tests/test_add_pushbover.py]: use `reverse`
avoid single letter variables such as `r`

File: hc/front/tests/test_add_channel.py [hunk 8]
```diff
@@ -38,4 +38,30 @@ def test_instructions_work(self):
...
+        self.assertContains(r, alice_channels.first().code)
+        self.assertEquals(r.status_code, 200)
+
+
+
     ### Test that bad kinds don't work
+    def test_bad_kinds_dont_work(self):
+        # Bad kinds means that a user can't add unsupported intergrations.
+        self.client.login(username="alice@example.org")
+        bad_kinds = ("whatsapp", "facebook", "linkedin")
```
Comment (johnseremba) [inline hc/front/tests/test_add_channel.py]: It is not necessary to loop in tests.
Just use one bad kind and test against it.

**Review by johnseremba — CHANGES_REQUESTED** [review]

File: hc/front/tests/test_add_channel.py (hunk 8 above)
Comment (samachola) [reply hc/front/tests/test_add_channel.py]: Noted.

File: hc/front/tests/test_add_channel.py (hunk 3 above)
Comment (samachola) [reply hc/front/tests/test_add_channel.py]: Noted.

File: hc/front/tests/test_add_channel.py (hunk 2 above)
Comment (samachola) [reply hc/front/tests/test_add_channel.py]: Noted

File: hc/front/tests/test_add_channel.py (hunk 1 above)
Comment (samachola) [reply hc/front/tests/test_add_channel.py]: Noted

Comment (LehruAngela) [reply hc/front/tests/test_add_channel.py]: Noted

**Review by johnseremba — APPROVED** [review]

**Pull request merged**